text.py

```diff
@@ -97,12 +97,13 @@ def viterbi_segment(text, P):
     n = len(text)
     words = [''] + list(text)
     best = [1.0] + [0.0] * n
-    # Fill in the vectors best, words via dynamic programming
+    # Fill in the vectors best words via dynamic programming
     for i in range(n+1):
         for j in range(0, i):
             w = text[j:i]
-            if P[w] * best[i - len(w)] >= best[i]:
-                best[i] = P[w] * best[i - len(w)]
+            curr_score = P[w] * best[i - len(w)]
```

Comment:
I have seen `curr_something` (or `currSomething` etc.) pop up a lot around, so I assume most people will get it. The problem with `p_score` is that `p` doesn't give enough information, which is not the same case as `curr_score`.

I personally feel it's fine, but I don't feel strongly about it. If more have the same issue, I will change it.